### PR TITLE
[api] Fix getFeaturesFromIds when GMF and WMS layers do not have the …

### PR DESCRIPTION
**Pull request by arnaud-morvan**
…same name

Need to be ported in master

### CODE REVIEW

File: api/src/Querent.js [hunk 1]
```diff
@@ -54,14 +54,17 @@ export function getFeaturesFromIds(layer, ids) {
...
+      }
 
       const params = {
         'FEATUREID': featureIds.join(','),
-        'MAXFEATURES': ids.length,
+        'MAXFEATURES': featureIds.length,
         'REQUEST': 'GetFeature',
         'SERVICE': 'WFS',
-        'TYPENAME': layer,
+        'TYPENAME': overlayDef.layer.layers,
```
Comment (arnaud-morvan) [inline api/src/Querent.js:67]: Use childLayers names instead

File: api/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import config from './src/constants.js';
 import Map from './src/Map.js';
 
 config.themesUrl = 'https://geomapfish-demo-2-4.camptocamp.com/themes?' +
-  'version=2&background=background&interface=api';
+  'version=2&background=background&interface=desktop';
```
Comment (sbrunner) [inline api/index.js:5]: This should not be committed...

File: api/src/Querent.js [hunk 3]
```diff
@@ -54,14 +54,19 @@ export function getFeaturesFromIds(layer, ids) {
         return;
       }
 
-      const featureIds = ids.map(id => `${layer}.${id}`);
+      const childLayerNames = [];
+      let featureIds = [];
+      for (const childLayer of overlayDef.layer.childLayers) {
```
Comment (sbrunner) [inline api/src/Querent.js:59]:
```suggestion
      for (const childLayer of overlayDef.childLayers) {
```

Comment (sbrunner) [reply api/src/Querent.js:59]: wrong - removing ...

Comment (sbrunner) [inline api/src/Querent.js:59]: ``overlayDef.layer`` should be casted into ``GmfLayerWMS``...

File: api/index.js (hunk 2 above)
Comment (arnaud-morvan) [reply api/index.js:5]: Commited by mistake, but note that interface API does not exists on demo 2.4.

File: api/src/Querent.js [hunk 4]
```diff
@@ -42,7 +42,7 @@ export function getFeaturesFromIds(layer, ids) {
 
       /** Array<import('ol/Feature.js').default> */
       let features = [];
-      const overlayDef = overlayDefs.get(layer);
+      const overlayDef = /** @type import('gmf/themes.js').GmfLayerWMS */ overlayDefs.get(layer);
```
Comment (sbrunner) [inline api/src/Querent.js:45]:
```suggestion
      const overlayDef = overlayDefs.get(layer);
      const gmfLayer = /** @type import('gmf/themes.js').GmfLayerWMS */ (overlayDef.layer);
```

File: api/index.js (hunk 2 above)
Comment (sbrunner) [reply api/index.js]: Just created :-)

**Comment by sbrunner** [comment]
Nice :-)

**Comment by arnaud-morvan** [comment]
Thanks @sbrunner

**Comment by sbrunner** [comment]
@arnaud-morvan do you see that the backports failed?

**Comment by arnaud-morvan** [comment]
Yes, already opened backport PRs. Master one is failing, but is it master branch which is broken.

**Comment by arnaud-morvan** [comment]
Sorry, master one is green now.

**Pull request merged**